nat-lab/tests/utils/network_switcher/network_switcher_windows.py

```diff
@@ -1,16 +1,68 @@
 import config
+import re
 from .network_switcher import NetworkSwitcher
+from dataclasses import dataclass
+from typing import List, Optional
 from utils.connection import Connection
 from utils.process import ProcessExecError
 
-VAGRANT_DEFAULT_IF = "Ethernet"  # 10.0.2.0/24
-PRIMARY_IF = "Ethernet 2"  # 10.55.0.11
-SECONDARY_IF = "Ethernet 3"  # 10.66.0.11
+
+@dataclass
+class Interface:
+    def __init__(self, name: str, ipv4: str) -> None:
+        self.name = name
+        self.ipv4 = ipv4
+
+
+class ConfiguredInterfaces:
+    def __init__(self, default: str, primary: str, secondary: str) -> None:
+        self.default = default
+        self.primary = primary
+        self.secondary = secondary
+
+    @staticmethod
+    async def find(connection: Connection) -> "ConfiguredInterfaces":
+        interfaces = await ConfiguredInterfaces.get_network_interfaces(connection)
+
+        def find_interface(prefix: str) -> str:
+            for interface in interfaces:
+                if interface.ipv4.startswith(prefix):
+                    return interface.name
+            assert False, f"interface not found with prefix `{prefix}`, {interfaces}"
+
+        return ConfiguredInterfaces(
+            find_interface(config.LIBVIRT_MANAGEMENT_NETWORK_PREFIX),
+            find_interface(config.PRIMARY_VM_NETWORK_PREFIX),
+            find_interface(config.SECONDARY_VM_NETWORK_PREFIX),
+        )
+
+    @staticmethod
+    async def get_network_interfaces(connection: Connection) -> List[Interface]:
+        process = await connection.create_process(
+            ["netsh", "interface", "ipv4", "show", "addresses"]
+        ).execute()
+
+        matches = re.findall(
+            r"Configuration for interface \"(.*)\"[\s\S]*?IP Address:\s*([\d.]*)",
+            process.get_stdout(),
+        )
+
+        result: List[Interface] = []
+        for match in matches:
+            result.append(Interface(match[0], match[1]))
+
+        return result
```

Comment:
IMHO it looks more like `Interface` static method?

Reply:
I didn't understand what you mean. `get_network_interfaces` should be renamed to `Interface`?

Reply:
No, I meant that it doesn't really do anything with `ConfiguredInterfaces` objects, but operates on `Interface` objects, so I guess it would have more reason to make it a static method of `Interface` class

Reply:
Ok I see your point. I can move `get_network_interfaces` into `Interface` class. What do you think about getting rid of `Interface` class altogether and simply using named tuple? I haven't tried Python's named tuples yet, but if they work well I could use that instead.

Reply:
Nevermind the tuple, its defined the same way as a dataclass, so it still looks like a separate class. I will follow your suggestion.